HW3/ClientTCPHTTP.java

```diff
@@ -4,34 +4,114 @@
-        // TODO: HTTP request message should have the following lines with a “\r\n” at the end of each line:
-        /*  The request line (hint: the URL should include a ‘/’ in front of the htm file name) The header line for the field “Host:”
-         *  The header line for the field “User-Agent:”
-         *  <empty line>
-         */
-
-        // TODO: Receive and interpret the HTTP response message from the HTTP Server program line by line
-        // TODO: Display the RTT (File Transmission Time may be included) of HTTP query in ms as a single line (e.g., RTT = 1.089 ms)
-        // TODO: Display the status line and header lines of the HTTP response message on the standard output
-        // TODO: Save the data in the entity body to a .htm file to local directory if there is any.
-
-        // TODO: Display a message on the standard output to ask the User whether to continue. If yes, repeat steps 3 through 6. Otherwise, close all i/o streams, TCP connection, and terminate the Client program.
-        System.out.println("This is the Client");
+    public static void main(String[] args) throws IOException {
+        Socket tcpSocket = null;
+        PrintWriter socketOut = null;
+        BufferedReader socketIn = null;
+        String host, fromUser, fromServer, httpMethod, htmFile, httpVersion, userAgent;
+        long beforeTCP, afterTCP, tcpRtt, beforeFT, afterFT, ftRTT;
+        int port = 5310; // 5140 Jesse, 5310 Alyssa
+
+        BufferedReader sysIn = new BufferedReader(new InputStreamReader(System.in));
+        // Ask user for DNS/IP of HTTP server
+        System.out.print("Please enter the DNS/IP address: ");
+        host = sysIn.readLine();
+
+        // Create TCP connection to HTTP server with the Host Name input by User at the
+        // given port
+        try {
+            beforeTCP = new Date().getTime();
+            tcpSocket = new Socket(host, port);
+            socketOut = new PrintWriter(tcpSocket.getOutputStream(), true);
+            socketIn = new BufferedReader(new InputStreamReader(tcpSocket.getInputStream()));
+            afterTCP = new Date().getTime();
+
+            // Display the RTT of TCP connection in ms(difference between the moments right
+            // before and after creating socket object).
+            tcpRtt = afterTCP - beforeTCP;
+            System.out.println("RTT of TCP connection: " + Long.toString(tcpRtt) + " ms");
+
+        } // Catch any exception, terminate the program, and display error messages to
+          // terminal.
+        catch (UnknownHostException e) {
+            System.err.println("Unknown Host: " + host);
+            System.exit(1);
+        } catch (IOException e) {
+            System.err.println("Couldn't get I/O for the connection to: " + host);
+            System.exit(1);
+        }
+        
+        while ((fromUser = sysIn.readLine()) != "N") {
```

Comment:
will need to see why the server is waiting on a keyboard input before it  moves on to next print statement

Reply:
fixed this in #15!